### PR TITLE
Added functional chat page and delete conversations endpoint

### PR DESCRIPTION
**Pull request by Y0z64**
Solves #29 

The query calls for the chat are not limited in any way shape or form, so actions as small as changing tabs will recall the entire conversation and load it again, which is not good for the performance.

I also made a mess of the url params when implementing the global chat, that needs to be fixed

- [x] Restrict chat query calls in some way
- [x] Add file references and reference count  
- [ ] Fix the global chat url params

Dont merge until all this points are solved

EDIT de Axel: Movi esto aca arriba para que salga el progreso de los ToDos

### CODE REVIEW

**Comment by Y0z64** [comment]
The query calls for the chat are not limited in any way shape or form, so actions as small as changing tabs will recall the entire conversation and load it again, which is not good for the performance.

- [ ] Restrict chat query calls in some way

**Review by Gorchon — COMMENTED** [review]
.'"'.        ___,,,___        .'``.
: (\  `."'"```         ```"'"-'  /) ;
 :  \                         `./  .'
  `.                            :.'
    /        _         _        \
   |         0}       {0         |
   |         /         \         |
   |        /           \        |
   |       /             \       |
    \     |      .-.      |     /
     `.   | . . /   \ . . |   .'
 jgs   `-._\.'.(     ).'./_.-'
           `\'  `._.'  '/'
             `. --'-- .'
               `-...-'

File: src/components/react/chat.tsx [hunk 1]
```diff
@@ -15,40 +17,71 @@ const Chat: React.FC<ChatProps> = (props) => {
...
-    queryFn: async () => {
+const InnerChat: React.FC<ChatProps> = ({ id, type, showRefs = false }) => {
+  const messagesRes = useQuery(
+    ["messages", type, id], // queryKey
+    async () => {
+      // queryFn
       const endpoint = id
         ? `/api/assistant/chat/${id}?type=${type}`
-        : `/api/assistant/chat`;
+        : `/api/assistant/chat/all`;
```
Comment (adpadillar) [inline src/components/react/chat.tsx]: Why do we use the same endpoint (/api/assistant/chat/[id]) for this, when that is not what we had planned in the project documents before. These should be two separate endpoints.
1. (already was done) /api/assistant/chat/[id] to chat with a specific file/note
2. /api/assistant/chat to chat with all

File: src/components/react/chat.tsx [hunk 2]
```diff
@@ -15,40 +17,71 @@ const Chat: React.FC<ChatProps> = (props) => {
...
       queryClient.invalidateQueries({ queryKey: ["messages", type, id] }),
+    onSuccess: () => {
+      console.log(sendMessageMutation.data?.referencedFiles);
+    },
+  });
+
+  const deleteConversationMutation = useMutation({
+    mutationKey: [id, "delete-conversation-mutation"],
+    mutationFn: async (id: string) => {
+      const endpoint = `/api/conversations/delete`;
```
Comment (adpadillar) [inline src/components/react/chat.tsx]: This should probably be DELETE /api/conversations/[id], no need to send body or specify /delete

File: src/pages/api/assistant/chat/[id].ts [hunk 3]
```diff
@@ -41,17 +47,23 @@ export const GET: APIRoute = async ({ locals, request, params }) => {
...
     });
   }
 
   let conversation = await db.query.conversations.findFirst({
-    where: and(
-      eq(conversations.userId, currentUser.id),
-      or(eq(conversations.fileId, id), eq(conversations.noteId, id))
-    ),
+    where:
+      id !== "all"
```
Comment (adpadillar) [inline src/pages/api/assistant/chat/[id].ts]: I feel like we should follow the structure we had in the document and split the endpoints with a note/file and the endpoints without

File: src/pages/api/files/delete.ts [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import type { APIRoute } from "astro";
```
Comment (adpadillar) [inline src/pages/api/files/delete.ts]: This endpoint should also probable be DELETE /api/files/[id], as we had in the original document

**Review by adpadillar — CHANGES_REQUESTED** [review]

**Comment by adpadillar** [comment]
Also, chat breaks when notes are returned as context (instead of files)

**Comment by Y0z64** [comment]
> Also, chat breaks when notes are returned as context (instead of files)

I thought you where only suposed to be able to talk with files so thats by bad

**Comment by adpadillar** [comment]
Just merged my PR on this one because I needed your changes. This is what I changed
- [x]  **added functionality to search on files AND notes**
- [x] Improve UI for search preview on notes result
- [x] Improve logic around choosing the file to preview, ideally ask the model what would be the best file to display

**Pull request merged**